src/main/java/com/meilisearch/sdk/Index.java

```diff
@@ -567,6 +567,40 @@ public Task resetDistinctAttributeSettings() throws Exception {
         return this.settingsHandler.resetDistinctAttributeSettings(this.uid);
     }
 
+    /**
+     * Get the typo tolerance field of an index.
+     * https://docs.meilisearch.com/reference/api/distinct_attribute.html#get-distinct-attribute
+     *
+     * @return typo tolerance field of a given uid as TypoTolerance instance
+     * @throws Exception if an error occurs
+     */
+    public TypoTolerance getTypoToleranceSettings() throws Exception {
+        return this.settingsHandler.getTypoToleranceSettings(this.uid);
+    }
+
+    /**
+     * Update the typo tolerance field of an index.
+     * https://docs.meilisearch.com/reference/api/distinct_attribute.html#update-distinct-attribute
+     *
+     * @param typoTolerance A TypoTolerance instance
+     * @return Task instance
+     * @throws Exception if an error occurs
```

Comment:
It throw an Exception like:
```bash
Exception in thread "main" MeiliSearchApiException{error=APIError{message='Index `movies` not found.', code='index_not_found', type='invalid_request', link='https://docs.meilisearch.com/errors#index_not_found'}}
```

Reply:
Yes, that's I'm talking about, why not defining like this: `throws MeiliSearchApiException` ?

Reply:
I can't I can change it for:
```java
     * @throws Exception if the client has an error
     * @throws MeiliSearchApiException if the response is an error
     */
    public TypoTolerance getTypoToleranceSettings() throws Exception, MeiliSearchApiException {
        return this.settingsHandler.getTypoToleranceSettings(this.uid);
    }
```
Or, I can throw it with a `try` `catch` block. I don't know how you want to do it?

Reply:
Sorry, I should be more clear about what I asked!

I mean, is good to only `throw` errors that we can control, `Exception` is Ok, but in the end is easier to the consumer just to expect a MeiliSearchApiException instead of the `Exception`, because they will have more control over the exception catch flow like this:

```java
// when we just throw a Meilisearch error, the user can catch all the Meilisearch errors first:
// of course, they need to respect an inheritance (we can talk about this later).

try {
  client.index("movies").anyMethodThatThrowsSomething();
} catch(MeiliSearchOtherException e) {
  // do something with a meilisearch error.
} catch(MeiliSearchActionException e) {
  // do something with a meilisearch error.
} catch(MeiliSearchApiException e) {
  // do something with a meilisearch error.
}
```

Now if we just throw an Exception:

```java
// in this new case, we can see the code is different and harder to read because they had to handle "multiple" 
// different exception possibilities or even check if this exception isn't coming from the okhttp for example.

try {
  client.index("movies").anyMethodThatThrowsSomething();
} catch(Exception e) {
   if (e.message.startsWith("message")) // do something
   else if (e.message.startsWith("other message")) // do other thing 
}
```